internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2531,6 +2568,19 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 	return resourceKubernetesClusterRead(d, meta)
 }
 
+func validateKubernetesClusterUpgradeOverrideSetting(_, new interface{}) error {
```

Comment:
Can we remove this function and make it inline in the `update` function? I think we should be able to use something like `d.GetRawConfig` so we could compare the `old` values of the fields against the `new` values of the fields.